File: backend/src/main/scala/domains/user/model/UpdateOwnSettingsRequest.scala
package domains.user.model

import domains.auth.model.{DisplayName, EmailAddress, PlaintextPassword}
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateOwnSettingsRequest(
  displayName: DisplayName,
  email: EmailAddress,
  preferences: UserPreferences,
  currentPassword: PlaintextPassword,
  newPassword: Option[PlaintextPassword]
)

object UpdateOwnSettingsRequest:
  given Encoder[UpdateOwnSettingsRequest] = deriveEncoder[UpdateOwnSettingsRequest]
  given Decoder[UpdateOwnSettingsRequest] = deriveDecoder[UpdateOwnSettingsRequest]

File: frontend/src/features/user/model/UpdateOwnSettingsRequest.ts
import type { DisplayName, EmailAddress, PlaintextPassword } from '@/features/auth/model/AuthValues'
import type { UserPreferences } from '@/features/user/model/UserPreferences'

export type UpdateOwnSettingsRequest = {
  displayName: DisplayName
  email: EmailAddress
  preferences: UserPreferences
  currentPassword: PlaintextPassword
  newPassword: PlaintextPassword | null
}
